test/currencyConvertor.ts

```diff
@@ -1,73 +1,196 @@
-    const signers = await ethers.getSigners();
+  let currencyConvertor: CurrencyConvertor;
+  let zeroExExchangeWrapper: ZeroExExchangeWrapper;
+  let usdcTokenContract: ERC20;
+  let usdtTokenContract: ERC20;
+  let signer: SignerWithAddress;
 
-    zeroExExchangeWrapper = await deployContract(
-      signers[0],
-      ZeroExExchangeWrapperArtifact,
-    );
-    starkware = await deployContract(
-      signers[0],
-      StarkwareArtifact,
-    );
-    usdc = await deployContract(
-      signers[0],
-      UsdcContractArtifact,
-    )
-
-    await usdc.mint(
-      zeroExExchangeWrapper.address,
-      1000e6,
-    )
-    await usdc.mint(
-      starkware.address,
-      1000e6,
-    )
+  before(async () => {
+    // setup account
+    await hre.network.provider.request({
+      method: "hardhat_impersonateAccount",
+      params: [impersonatedAccount],
+    });
+    await hre.network.provider.request({
+      method: "hardhat_setBalance",
+      params: [
+        impersonatedAccount,
+        "0x1000000000000000000",
+      ],
+    });
+    signer = await ethers.getSigner(impersonatedAccount);
 
     currencyConvertor = await deployContract(
-      signers[0],
+      signer,
       CurrencyConvertorArtifact,
+    ) as CurrencyConvertor;
+    zeroExExchangeWrapper = await deployContract(
+      signer,
+      ZeroExExchangeArtifact,
       [
-        starkware.address,
-        usdc.address, // placeholder for USDC
-        156,
+        starkwareContractAddress,
+        usdcAddress,
+        '0x02893294412a4c8f915f75892b395ebbf6859ec246ec365c3b1f56f47c3a0a5d',
       ],
-    );
+    ) as ZeroExExchangeWrapper;
 
-    await usdc.mint(
-      signers[0].address,
-      1000e6,
-    )
+    // get ERC20 contracts
+    usdtTokenContract = new ethers.Contract(
+      usdtTokenAddress,
+      erc20Abi,
+      signer,
+    ) as ERC20;
+    usdcTokenContract = new ethers.Contract(
+      usdcAddress,
+      erc20Abi,
+      signer,
+    ) as ERC20;
 
-    await usdc.approve(currencyConvertor.address, 1e10);
+    // approve ERC20 contracts
+    await usdtTokenContract.approve(
+      currencyConvertor.address,
+      100000000000,
+    );
   });
+
+  after(async () => {
+    await hre.network.provider.request({
+      method: "hardhat_stopImpersonatingAccount",
+      params: [impersonatedAccount],
+    });
+  })
+
   describe("deposit", async () => {
-    it("deposit USDT to USDC", async () => {
+    it("deposit USDT as USDC to Starkware", async () => {
+      const zeroExTransaction = await zeroExRequest('100');
+
+      // get old balances
+      const userUsdtBalance: BigNumber = await usdtTokenContract.balanceOf(signer.address);
+      const starkwareUsdcBalance: BigNumber = await usdcTokenContract.balanceOf(
+        starkwareContractAddress,
+      )
+
+      await zeroExExchangeWrapper.approveSwap(zeroExTransaction.to, usdtTokenAddress);
+      const tx = await currencyConvertor.deposit(
+        usdtTokenAddress,
+        '100000',
```

Comment:
Nit: use `parseUnits` instead of hardcoding: https://docs.ethers.io/v5/api/utils/display-logic/#utils-parseUnits

Reply:
sorry I dont understand this comment - i am passing in the max amount of USDT willing to be given up